ext/scheduler/airflow2/resources/__lib.py

```diff
@@ -355,44 +355,43 @@ def poke(self, context):
         # TODO this needs to be updated to use optimus get job spec
         upstream_schedule = self.get_schedule_interval(schedule_time)
 
-        _, last_upstream_execution_date = SuperExternalTaskSensor.get_last_upstream_times(
+        last_upstream_schedule_time, _ = SuperExternalTaskSensor.get_last_upstream_times(
             schedule_time,
             upstream_schedule)
 
         # get schedule window
         task_window = JobSpecTaskWindow(self.window_size, 0, "m", self._optimus_client)
-        execution_date_window_start, execution_date_window_end = task_window.get(
-            last_upstream_execution_date.strftime(TIMESTAMP_FORMAT))
-
-        self.log.info(
-            "upstream interval: {}, window size: {}".format(upstream_schedule, self.window_size))
-        self.log.info(
-            "waiting for upstream runs between: {} - {} execution dates of airflow dag run".format(
-                execution_date_window_start.isoformat(), execution_date_window_end.isoformat()))
-
-        # find success iterations we need in window
-        expected_upstream_executions = SuperExternalTaskSensor.get_expected_upstream_executions(upstream_schedule,
-                                                                                                execution_date_window_start,
-                                                                                                execution_date_window_end)
-        self.log.info("expected upstream executions ({}): {}".format(len(expected_upstream_executions),
-                                                                     expected_upstream_executions))
+        schedule_time_window_start, schedule_time_window_end = task_window.get(
+            last_upstream_schedule_time.strftime(TIMESTAMP_FORMAT))
+        
+        schedule_time_window_start_next, schedule_time_window_end = self.get_inclusive_start_schedule_window(
```

Comment:
it is better to move get_inclusive_start_schedule_window  method under JobSpecTaskWindow; and we can avoid calling task_window.get; and rather call get_schedule_window which internally call task_window.get